### PR TITLE
[Data Sprint] Track products orders and payments plugin once per site login per installation

### LINKED ISSUE #9588: Track products, orders and payments plugin once per site login per installation
Context - pdND4G-gg-p2#comment-244

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #9588

### Description

The PR adds an event, that tracks:
* orders count
* products count
* payments gateways status
It's tracked once per the app installation per used site

More context - pdND4G-gg-p2#comment-243

### Testing instructions

* use logcat with "track" filtering
* run the app
* Notice the tracked event. e.g.
```
Tracked: application_store_snapshot, Properties: {"products_count":1078,"orders_count":778,"woocommerce-payments":"installed_and_activated","woocommerce-gateway-stripe":"not_installed","woocommerce-square":"installed_and_not_activated","woocommerce-paypal-payments":"installed_and_activated","blog_id":192152755,"is_wpcom_store":true,"was_ecommerce_trial":false,"plan_product_slug":"business-bundle","is_debug":true}
```

* Change site and close/open the app. Notice that it's tracked again for the new site
* Close/open without changing site - notice that it's not tracked

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/androidTest/assets/mocks/mappings/jetpack-blogs/wc/products/reports_products_totals.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (kidinov) [inline WooCommerce/src/androidTest/assets/mocks/mappings/jetpack-blogs/wc/products/reports_products_totals.json:1]: @woocommerce/mobile-ui-testing-squad  👋

I was wondering, can we just put "false" to `TestBase::43` or is there something behind the fact that we need to have stubs for all the requests that are called during tests in `e2e` package?

**Review by samiuelson — APPROVED** [review]
Tests good!

File: WooCommerce/src/androidTest/assets/mocks/mappings/jetpack-blogs/wc/products/reports_products_totals.json (hunk 1 above)
Comment (pachlava) [reply WooCommerce/src/androidTest/assets/mocks/mappings/jetpack-blogs/wc/products/reports_products_totals.json:1]: Sorry for the late reply @kidinov, I was AFK for a week.

This is a good question. In fact, this is a conscious decision and a sort of experiment. For Android, any unmatched request will trigger a test fail. For iOS, unmatched requests warnings from WireMock are ignored, so the test will fail only if a mocked API so far behind the real one, that it no longer matches what's expected by the UI.

I'd say, it's less painful to keep the mocks in order (as we are "forced" to do with Android), than to get into a pitfall of dozens of unmatched requests while figuring out which one exactly is causing a fail (as often happens with iOS).

This approach (for both platforms) is not carved in stone, and can be changed, though.

Comment (kidinov) [reply WooCommerce/src/androidTest/assets/mocks/mappings/jetpack-blogs/wc/products/reports_products_totals.json:1]: thanks @pachlava. That makes a lot of sense to me

**Pull request merged**